src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java

```diff
@@ -415,6 +415,16 @@ private static boolean isSingleLineLambda(DetailAST lambda) {
         final DetailAST block = lambda.getLastChild();
         if (block.getType() != TokenTypes.SLIST) {
             result = lambda.getLineNo() == block.getLineNo();
+            final DetailAST elist = block.getFirstChild().findFirstToken(TokenTypes.ELIST);
+            if (elist != null) {
+                result = result && block.getLineNo() == elist.getLastChild().getLineNo();
```

Comment:
`result = result && ....` looks weird, please put check for `result == false` in IF and skip all this block completely.